File: DSA_python/DSA03010 NỐI DÂY 1.py
import heapq

def submit():
    n = int(input())
    A = list(map(int, input().split()))
    heapq.heapify(A)

    total, temp = 0, 0
    while len(A) != 1:
        temp += A[0]
        heapq.heappop(A)
        temp += A[0]
        heapq.heappop(A)

        total += temp
        heapq.heappush(A, temp)
        temp = 0
    
    print(total)

test = int(input())
for i in range(test):
    submit()

File: DSA/DSA03010 NỐI DÂY 1.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    priority_queue<int, vector<int>, greater<int>> A;

    for (int i = 0; i < n; i++)
    {
        int temp;
        cin >> temp;
        A.push(temp);
    }
    
    long long total = 0, temp = 0;
    while (A.size() > 1)
    {
        temp += A.top(); A.pop();
        temp += A.top(); A.pop();

        total += temp;
        A.push(temp);
        temp = 0;
    }
    
    cout << total;

    cout << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int test;
    cin >> test;
    while (test--)
    {
        submit();
    }
    

    return 0;
}